core/src/services/hdfs_native/lister.rs

```diff
@@ -15,28 +15,62 @@
 // specific language governing permissions and limitations
 // under the License.
 
-use std::sync::Arc;
+use chrono::DateTime;
+use hdfs_native::client::{FileStatus, ListStatusIterator};
 
-use crate::raw::oio;
 use crate::raw::oio::Entry;
+use crate::raw::{build_rel_path, oio};
+use crate::services::hdfs_native::error::parse_hdfs_error;
 use crate::*;
 
 pub struct HdfsNativeLister {
-    _path: String,
-    _client: Arc<hdfs_native::Client>,
+    root: String,
+    lsi: ListStatusIterator,
 }
 
 impl HdfsNativeLister {
-    pub fn new(path: String, client: Arc<hdfs_native::Client>) -> Self {
-        HdfsNativeLister {
-            _path: path,
-            _client: client,
+    pub fn new(root: &str, lsi: ListStatusIterator) -> Self {
+        Self {
+            root: root.to_string(),
+            lsi,
         }
     }
 }
 
 impl oio::List for HdfsNativeLister {
     async fn next(&mut self) -> Result<Option<Entry>> {
-        todo!()
+        let de: FileStatus = match self.lsi.next().await {
+            Some(res) => match res {
+                Ok(fs) => fs,
+                Err(e) => return Err(parse_hdfs_error(e)),
+            },
+            None => return Ok(None),
+        };
+
+        let path = build_rel_path(&self.root, &de.path);
+
+        let entry = if !de.isdir {
+            let odt = DateTime::from_timestamp(de.modification_time as i64, 0);
+            let dt = match odt {
```

Comment:
Please don't `match` an `Option` if the `None` case returns an error, use `let-else` instead.